backend/modules/auth/service.py

```
@@ -1,47 +1,158 @@
+        ).first()
+        
+        if existing_user:
+            if existing_user.email == email:
+                raise HTTPException(status_code=400, detail="Email already registered")
+            else:
+                raise HTTPException(status_code=400, detail="Username already taken")
+        
+        # Generate role-based user ID
+        try:
+            user_id = generate_unique_user_id(db, role)
+        except Exception as e:
```

Ruff (0.14.6)

Do not catch blind exception: `Exception`

(BLE001)